packages/flutter/lib/src/widgets/overlay.dart

```diff
@@ -81,7 +83,11 @@ class OverlayEntry implements Listenable {
     bool opaque = false,
     bool maintainState = false,
   }) : _opaque = opaque,
-       _maintainState = maintainState;
+       _maintainState = maintainState {
+    if (kFlutterMemoryAllocationsEnabled) {
+      _maybeDispatchObjectCreation();
```

Comment:
@polina-c  Why is if(kFlutterMemoryAllocationsEnabled) being checked twice, once before invoking the _maybeDispatchObjectCreation function and one time within it? Is there a specific reason for this double-check, or am I missing something in this context?